server/request_pusher.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/flashbots/rpc-endpoint/database"
+	"time"
+)
+
+type RequestPusher struct {
+	db         database.Store
+	EntryChan  chan database.Entry
+	BatchChan  chan []database.Entry
+	maxItem    int           // maximum item to be pushed as batch
+	maxTimeOut time.Duration // timeout in which entry to be added to batch queue
+}
+
+func NewRequestPusher(db database.Store, maxItem int, maxTimeOut time.Duration) *RequestPusher {
+	return &RequestPusher{
+		db:         db,
+		EntryChan:  make(chan database.Entry, 100),
+		BatchChan:  make(chan []database.Entry),
+		maxItem:    maxItem,
+		maxTimeOut: maxTimeOut,
+	}
+}
+
+func (r *RequestPusher) Run() {
+	go r.saveEntries()
+	go r.addEntryToEntryQueue()
+}
+
+func (r *RequestPusher) addEntryToEntryQueue() {
+	for keepGoing := true; keepGoing; {
+		var batch []database.Entry
+		ticker := time.NewTicker(r.maxTimeOut)
```

Comment:
We've seen issues when creating timers in a loop in the past - they are not cleaned up properly or fast enough and cause resource bleed resulting in scheduler running too slow (~80% cpu spent in the scheduler iterating old timers).
You could declare the ticker outside and reset it instead of creating a new one each iteration.

Reply:
ok.makes sense.